vector/src/main/java/im/vector/app/features/call/transfer/CallTransferViewModel.kt

```diff
@@ -22,22 +22,16 @@ import dagger.assisted.AssistedFactory
 import dagger.assisted.AssistedInject
 import im.vector.app.core.di.MavericksAssistedViewModelFactory
 import im.vector.app.core.di.hiltMavericksViewModelFactory
-import im.vector.app.core.extensions.exhaustive
+import im.vector.app.core.platform.EmptyAction
 import im.vector.app.core.platform.VectorViewModel
-import im.vector.app.features.call.dialpad.DialPadLookup
 import im.vector.app.features.call.webrtc.WebRtcCall
 import im.vector.app.features.call.webrtc.WebRtcCallManager
-import im.vector.app.features.createdirect.DirectRoomHelper
-import kotlinx.coroutines.launch
-import org.matrix.android.sdk.api.extensions.orFalse
 import org.matrix.android.sdk.api.session.call.CallState
 import org.matrix.android.sdk.api.session.call.MxCall
 
 class CallTransferViewModel @AssistedInject constructor(@Assisted initialState: CallTransferViewState,
```

Comment:
is this class still needed? without the handle method it looks like it doesn't do much!

Reply:
Yea doesn't do much but it still listens for the call completion and sends a `CallTransferViewEvents.Complete`, which is still required and feels like correct place for this. So I would tend to favour consistency and what somebody reading for the first time might find expectable vs optimising for less classes. But totally happy to put it somewhere else if it isn't shoehorned. :)

Reply:
no strong opinions from my side, would agree that using a consistent pattern outweighs the benefit of removing a little bit of boilerplate